encrypt/ibe/ibe_test.go

```diff
@@ -12,92 +12,196 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func newSetting() (pairing.Suite, kyber.Point, []byte, kyber.Point) {
+func newSetting(i uint) (
+	pairing.Suite, kyber.Point, []byte, kyber.Point,
+	func(s pairing.Suite, master kyber.Point, ID []byte, msg []byte) (*Ciphertext, error),
+	func(s pairing.Suite, private kyber.Point, c *Ciphertext) ([]byte, error),
```

Comment:
it looks like you already made the hard work of putting things as generic as possible here - right ? Why not extending to the implementation ? It's a bit sad to have "two functions that does the same thing", while we already have generics "group interface". Anyway you see possible ?

Reply:
I guess we could have a generic `Encrypt(s pairing.Suite, master kyber.Point, ID []byte, msg []byte) (*Ciphertext, error)` functions relying on the other function according to the type of the `master kyber.Point`...

WDYT?

Reply:
Okay, this isn't so simple actually. It's easier to do in tlock than here because we don't really have a type for G1 and G2, so not that easy to match on it without an instantiation like we have in kyber-bls12381.